app/models/forms/condition_computer.rb

```diff
@@ -0,0 +1,69 @@
+# Computes display conditions that are implied from SkipRules.
+module Forms
+  class ConditionComputer
+    attr_accessor :form, :form_items, :table, :active_rules, :last_item_cache
+
+    def initialize(form)
+      self.form = form
+      self.form_items = form.preordered_items(eager_load: [:display_conditions, skip_rules: :conditions])
+    end
+
+    def condition_group_for(item)
+      table[item] || empty_group
+    end
+
+    def build_table
+      self.table = {}
+      self.active_rules = []
+      self.last_item_cache = {}
+      form_items.each do |item|
+        add_display_conditions(item)
```

Comment:
Doing this work in the same loop depends on the items being preordered, correct? I think a comment about that would be helpful for the future.